packages/snaps-simulator/src/features/configuration/Configuration.tsx

```diff
@@ -14,34 +14,97 @@ import {
   Textarea,
   Text,
   Divider,
+  InputLeftAddon,
+  InputRightAddon,
+  InputGroup,
+  Select,
 } from '@chakra-ui/react';
-import { FormEvent } from 'react';
+import { FormEvent, useEffect, useState } from 'react';
 
 import { useDispatch, useSelector } from '../../hooks';
 import {
   getOpen,
   getSesEnabled,
-  getSnapUrl,
+  getSnapId,
+  getSnapVersion,
   getSrp,
   setOpen,
-  setSnapUrl,
+  setSnapId,
+  setSnapVersion,
 } from './slice';
+import {
+  SnapIdPrefixes,
+  getSnapPrefix,
+  stripSnapPrefix,
+} from '@metamask/snaps-utils';
+import {
+  fetchNpmMetadata,
+  DEFAULT_NPM_REGISTRY,
+} from '@metamask/snaps-controllers/dist/snaps/location';
 
 export const Configuration = () => {
   const dispatch = useDispatch();
-  const snapUrl = useSelector(getSnapUrl);
+  const snapUrl = useSelector(getSnapId);
+  const snapVersion = useSelector(getSnapVersion);
   const srp = useSelector(getSrp);
   const sesEnabled = useSelector(getSesEnabled);
   const isOpen = useSelector(getOpen);
 
+  const [snapIdInput, setSnapIdInput] = useState(stripSnapPrefix(snapUrl));
+  const [snapIdPrefix, setSnapIdPrefix] = useState(getSnapPrefix(snapUrl));
+  const [npmVersions, setNpmVersions] = useState<string[]>([]);
+  const [selectedNpmVersion, setSelectedNpmVersion] = useState<
+    string | undefined
+  >(snapVersion);
+
+  const isNPM = snapIdPrefix === SnapIdPrefixes.npm.slice(0, -1);
+
   const handleClose = () => {
+    dispatch(setSnapVersion(selectedNpmVersion));
+    dispatch(setSnapId(`${snapIdPrefix}:${snapIdInput}`));
     dispatch(setOpen(false));
   };
 
+  const handleSnapPrefixChange = (event: FormEvent<HTMLSelectElement>) => {
+    setSnapIdPrefix(event.currentTarget.value as SnapIdPrefixes);
+  };
+
   const handleSnapUrlChange = (event: FormEvent<HTMLInputElement>) => {
-    dispatch(setSnapUrl(event.currentTarget.value));
+    setSnapIdInput(event.currentTarget.value);
   };
 
+  const handleNpmVersionChange = (event: FormEvent<HTMLSelectElement>) => {
+    setSelectedNpmVersion(event.currentTarget.value);
+  };
+
+  useEffect(() => {
+    let cancelled = false;
```

Comment:
Out of scope for this, but it would be cool if we could attach an abort controller to any fetch requests and cancel them.

Reply:
Yeah